devdoc/channel.md

```diff
@@ -0,0 +1,297 @@
+# `channel` requirements
+
+## Overview
+
+`channel` is a module that allows the user to move data asynchronously from producer to consumer without having to block the thread.
+
+Consumers of data call `channel_pull` to register a `PULL_CALLBACK` function to be called when data becomes available.
+
+Producers of data call `channel_push` to notify the channel that data is available and provide a `PUSH_CALLBACK` to be called when the data has been consumed.
+
+`channel_pull` and `channel_push` can be called from different threads in any order. The callbacks are executed on threadpool threads.
+
+## `channel` internal state
+
+The internal state of the `channel` module can have the following values:
+  - `CHANNEL_EMPTY`: This is the initial state of the `channel`. It means that there is no data available to be pulled. Calling `channel_pull` or `channel_push` will cause the state to transition to `CHANNEL_PUSHING` or `CHANNEL_PULLING` respectively.
```

Comment:
does this internal state ever return back to "EMPTY"?

What happens after push+pull+callbacks executed? What's the state now?

Reply:
Yes it returns back to EMPTY. Added explanation.